opencog/query/PatternMatchEngine.cc

```diff
@@ -57,9 +57,10 @@ static inline bool log(const Handle& h)
 
 static inline void logmsg(const char * msg, const Handle& h)
 {
-	LAZY_LOG_FINE << msg << (h == nullptr ?
-	                         std::string("(invalid handle)") :
-	                         h->toShortString());
+	LAZY_LOG_FINE << msg << std::endl
```

Comment:
It's exactly for readability reason I've put this newline, as otherwise it wrecks the indentation of h (when h is a link, which happens often). But after a second thought I agree its a bad idea, it's better to let the user decide by adding \n in the msg.